shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_shim_tracer.py

```diff
@@ -0,0 +1,90 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import wrapt
+from opencensus.trace.blank_span import BlankSpan
+from opencensus.trace.tracers.base import Tracer as BaseTracer
+
+from opentelemetry import context, trace
+from opentelemetry.shim.opencensus._shim_span import ShimSpan
+
+_logger = logging.getLogger(__name__)
+
+_SHIM_SPAN_KEY = context.create_key("opencensus-shim-span-key")
+
+
+def set_shim_span_in_context(
+    span: ShimSpan, ctx: context.Context
+) -> context.Context:
+    return context.set_value(_SHIM_SPAN_KEY, span, ctx)
+
+
+def get_shim_span_in_context() -> ShimSpan:
+    return context.get_value(_SHIM_SPAN_KEY)
+
+
+# pylint: disable=abstract-method
+class ShimTracer(wrapt.ObjectProxy):
+    def __init__(
+        self, wrapped: BaseTracer, *, otel_tracer: trace.Tracer
```

Comment:
I'm a bit confused of the purpose of `ObjectProxy`. My understanding is that it is an easy way for `ShimTracer` to get all the properties/methods of the `wrapped` object as well as pass them to it. Is this correct?

Reply:
Explained a bit in a different comment, but yes it falls back to the wrapped object if the actual ShimTracer does not define a field/method. Wrapt docs [here](https://wrapt.readthedocs.io/en/latest/wrappers.html#custom-object-proxies)

The goal is to use the `NoopTracer` implementation if the shim didn't implement something so that the shim is robust and future proof. For example, I didn't implement `finish()` so a call would pass-through to [this no-op impl](https://github.com/census-instrumentation/opencensus-python/blob/fd064f438c5e490d25b004ee2545be55d2e28679/opencensus/trace/tracers/noop_tracer.py#L32-L34).